### DISCUSSION TITLE
refresh relationship picker or other elements

### DISCUSSION THREAD

**Discussion by immohacker** [Budibase/budibase#12654] [Help]
Hi,
I dont know how can i do this.
Data Provider refresh do not work.

Her an example.
Ive have an relationship to an table called bankkarten.
but thebankcard i wish to use was not created yet.
So i have an sidepanel to create an new bankcard

Screenshot 1 the list. Our wished bankcard with vvv is not in list, of course, its not created yet

![Auswahl_1141](https://github.com/Budibase/budibase/assets/109019584/683e0c02-b7eb-4c4e-a890-c0d9fb1cb729)

Screenshot 2, we clicked on Bakkarte erstellen and sidepanel go up
![Auswahl_1142](https://github.com/Budibase/budibase/assets/109019584/f0d4382f-aace-4be0-a910-044eb1cae4d0)

Schreenshot 3, sidepanel with our new data, click on save.
here we created the bankcard vvv
![Auswahl_1143](https://github.com/Budibase/budibase/assets/109019584/ada6f67e-336b-4e61-a6b3-02396c6feb36)

and now we see, the new vvv is not in list.
On the save button i have an action to refresh the data provider.
Ive created an button for refresh dataprovider to.
nothing worked
But no vvv in list.
If i press F5 in the browser then the list refreshed, but all other fields i filled are gone of course due refrsh.
So i would like to refresh single fields

![Auswahl_1144](https://github.com/Budibase/budibase/assets/109019584/ac2e16d1-2c99-4891-9996-4a0e1a9b0135)
no vvv entry

I dont know what i make wrong or i can solve the Problem
Thank you!

Best regards Bonkersdeluxe

**Comment by leolivier** [comment]
I have exactly the same (maybe newbie) question. I'm using the internal budibase database.
I have a Person table and an Address Table as several people can live in the same address in my use case and I don't want to force people to reenter several times the address.
So I have a relationship one to many between Address and Person and in a Person form, I have a relationship field for the Address:
![image](https://github.com/Budibase/budibase/assets/4014752/a8f6e640-cff3-49d7-92d3-bb542c344d87)
and if the address does not already exist, I added a "new address" button which opens a modal form where I can create a new address. But when I close the modal form, the list of addresses is not updated and I can't find the address in the list...

**Reply by leolivier** [reply]
I found a bypass by using the TypeAhead plugin for the address and force users to search an address by some criteria.
But that's a bit cumbersome...

**Comment by melohagan** [comment]
It may be because you need to use the **Refresh Data Provider** action. Because you are using side panels, you will need to re-jig the components in your tree a bit to make sure the side panel (and form contents) sit nested under the Data provider that you want to refresh

**Reply by leolivier** [reply]
I'm not using side panel, the new address button opens a form in modal mode and I don't know how to do a refresh data provider from there.
But maybe the solution is precisely to use a side panel for that use case?

**Reply by melohagan** [reply]
Generally speaking we did add side panels as an improvement over modals. Could be worth giving them a go!

**Comment by immohacker** [comment]
Hi melohagan and leolivier,
Okay i tried this.
But it dont work.
What make i wrong, i show what ive build :

This is the relationship field, wich i will refresh after insert an new item over sidepanel.
At Top are the Dataprovider, so the relationship field is under the Dataprovider

![Auswahl_289](https://github.com/Budibase/budibase/assets/109019584/796ce7ab-47f3-45ce-81cd-16b9a6482dbc)
Here are the section were the list is. I put an button under the field, which opens the sidepanel.
![Auswahl_290](https://github.com/Budibase/budibase/assets/109019584/ae91297e-54d8-4cca-a0f1-8d534edf6dfe)

So we have thhis order:

The first parent ist Dataprovider, the next child is container gBankkarte, then the next child is the relationship field.
The second child from gBankkarte is the sidepanel, so the relationship field and the sidepanel sidepanelbankkarteerstellen are siblings and they parents are gBankkarte.
![Auswahl_291](https://github.com/Budibase/budibase/assets/109019584/c0ca685c-8239-4d10-ad51-f648e2a1dcf1)

In the sidepanel action at create there are the following actions:
![Auswahl_292](https://github.com/Budibase/budibase/assets/109019584/d6779e0a-f00a-40d2-9f5b-b3c202b14444)
![Auswahl_293](https://github.com/Budibase/budibase/assets/109019584/91ce9249-844f-4670-b51e-8ddb6fa84e0e)
But it doensnt work,
So i decided to add an button to refresh the list
But this doesnt work too
![Auswahl_294](https://github.com/Budibase/budibase/assets/109019584/08483f08-3f6a-4564-b6f1-9d993ae16b97)
The action from button
![Auswahl_295](https://github.com/Budibase/budibase/assets/109019584/4b12b063-1bd2-4e4c-ae5a-0331ae3e5765)
What should i do, only complete refresh via F5 key refreshed it, but all other filled fields will clear of course due refresh.
My goal is, to refresh the relationship field.
I hope you can help me.
Thank you very much!
Best regards Bonkersdeluxe

**Comment by immohacker** [comment]
Hi at all
no idea?
It  would be nice if that could be fixed or to clear what i make wrong.
thank you!
best regards Bonkersdeluxe

**Reply by melohagan** [reply]
Hey @immohacker The relationship picker does source its own data. I thought refreshing a parent data provider might trigger the relationship picker to refresh, but that does not appear to be the case.

If your relationship picker had a filter binding, then it would refresh upon change of that value. You could explore that, e.g. https://docs.budibase.com/docs/cascading-dropdown-filters

**Comment by immohacker** [comment]
Hi melohagan,
I refresh the data provider with the button, but the relationship picker dont refresh.
Or what you mean with :
I thought refreshing a parent data provider might trigger the relationship picker to refresh, but that does not appear to be the case.
Thank you!
Best regards Bonkersdeluxe
